Missing case in ExpressionTranslator

The following program crashes Silicon

```
domain foo  {
  function bar(): Bool 
  
  axiom {
    asserting (true) in true
  }
}

function baz(): Int
  requires bar()
```

<details><summary>Stack trace</summary>

```
Caused by: scala.MatchError: (asserting (true) in true) (of class viper.silver.ast.Asserting)                                                                                                                                                                   
        at viper.silicon.supporters.ExpressionTranslator.translate(ExpressionTranslator.scala:64)                                                                                                                                                               
        at viper.silicon.supporters.ExpressionTranslator.translate$(ExpressionTranslator.scala:24)                                                                                                                                                              
        at viper.silicon.supporters.DefaultDomainsTranslator.translate(Domains.scala:141)                                                                                                                                                                       
        at viper.silicon.supporters.DefaultDomainsTranslator.translateAxiom(Domains.scala:147)                                                                                                                                                                  
        at viper.silicon.supporters.DefaultDomainsTranslator.translateAxiom(Domains.scala:141)                                                                                                                                                                  
        at viper.silicon.supporters.DefaultDomainsContributor.$anonfun$collectDomainMembers$4(Domains.scala:102)                                                                                                                                                
        at viper.silicon.supporters.DefaultDomainsContributor.$anonfun$collectDomainMembers$4$adapted(Domains.scala:101)                                                                                                                                        
        at scala.collection.immutable.List.foreach(List.scala:333)                                                                                                                                                                                              
        at viper.silicon.supporters.DefaultDomainsContributor.$anonfun$collectDomainMembers$1(Domains.scala:101)                                                                                                                                                
        at viper.silicon.supporters.DefaultDomainsContributor.$anonfun$collectDomainMembers$1$adapted(Domains.scala:85)                                                                                                                                         
        at scala.collection.immutable.Set$Set1.foreach(Set.scala:168)                                                                                                                                                                                           
        at viper.silicon.supporters.DefaultDomainsContributor.collectDomainMembers(Domains.scala:85)
        at viper.silicon.supporters.DefaultDomainsContributor.analyze(Domains.scala:59)
        at viper.silicon.verifier.DefaultMainVerifier.$anonfun$analyzeProgramAndEmitPreambleContributions$1(DefaultMainVerifier.scala:539)
        at viper.silicon.verifier.DefaultMainVerifier.$anonfun$analyzeProgramAndEmitPreambleContributions$1$adapted(DefaultMainVerifier.scala:538)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at viper.silicon.verifier.DefaultMainVerifier.analyzeProgramAndEmitPreambleContributions(DefaultMainVerifier.scala:538)
        at viper.silicon.verifier.DefaultMainVerifier.verify(DefaultMainVerifier.scala:214)
        at viper.silicon.Silicon.viper$silicon$Silicon$$runVerifier(Silicon.scala:245)
        at viper.silicon.Silicon$$anon$1.call(Silicon.scala:199) 
        at viper.silicon.Silicon$$anon$1.call(Silicon.scala:198) 
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

</details>

--- a/src/main/scala/viper/silver/ast/Program.scala
+++ b/src/main/scala/viper/silver/ast/Program.scala
@@ -568,6 +568,7 @@ sealed trait DomainAxiom extends DomainMember {
     (if(!Consistency.noResult(exp)) Seq(ConsistencyError("Axioms can never contain result variables.", exp.pos)) else Seq()) ++
     (if(!Consistency.noOld(exp)) Seq(ConsistencyError("Axioms can never contain old expressions.", exp.pos)) else Seq()) ++
     (if(!Consistency.noLocationAccesses(exp)) Seq(ConsistencyError("Axioms can never contain location accesses.", exp.pos)) else Seq()) ++
+    (if(!Consistency.noAsserting(exp)) Seq(ConsistencyError("Axioms can never contain asserting expressions.", exp.pos)) else Seq()) ++
     (if(!(exp isSubtype Bool)) Seq(ConsistencyError("Axioms must be of Bool type", exp.pos)) else Seq()) ++
     Consistency.checkPure(exp)
 

--- a/src/main/scala/viper/silver/ast/utility/Consistency.scala
+++ b/src/main/scala/viper/silver/ast/utility/Consistency.scala
@@ -90,6 +90,9 @@ object Consistency {
   /** Returns true if the given node contains no location accesses. */
   def noLocationAccesses(n: Node) = !n.existsDefined { case _: LocationAccess => }
 
+  /** Returns true if the given node contains no asserting expressions. */
+  def noAsserting(n: Node) = !n.existsDefined { case _: Asserting => }
+
   /** Returns true if the given node contains no accessibility predicates (unfolding predicates is
     * allowed) and no magic wands (applying wands is allowed).
     */

--- a/src/test/resources/all/issues/silicon/0911.vpr
+++ b/src/test/resources/all/issues/silicon/0911.vpr
@@ -1,0 +1,15 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+
+domain foo  {
+  function bar(): Bool
+
+  axiom {
+    //:: ExpectedOutput(consistency.error)
+    asserting (true) in true
+  }
+}
+
+function baz(): Int
+  requires bar()

